romancal/resample/resample.py

```diff
@@ -732,30 +735,79 @@ def gwcs_into_l3(model, wcsinfo):
-    l3_wcsinfo.dec_ref = transform.lat_6.value
-    l3_wcsinfo.ra_ref = transform.lon_6.value
-    # l3_wcsinfo.x_ref = center of mosaic?
-    # l3_wcsinfo.y_ref = center of mosaic?
-    # l3_wcsinfo.s_region = from gwcs.bounding_box
-
-    # Mosaic info
-    # l3_wcsinfo.dec_center = from center of mosaic
-    # l3_wcsinfo.pixel_scale = ???
-    # l3_wcsinfo.pixel_scale_local = ???
-    # l3_wcsinfo.pixel_shape = image shape
-    # l3_wcsinfo.ra_center = from center of mosaic
-    # l3_wcsinfo.ra_corn1 = from bounding box
-    # l3_wcsinfo.ra_corn2 = from bounding box
-    # l3_wcsinfo.ra_corn3 = from bounding box
-    # l3_wcsinfo.ra_corn4 = from bounding box
-    # l3_wcsinfo.dec_corn1 = from bounding box
-    # l3_wcsinfo.dec_corn2 = from bounding box
-    # l3_wcsinfo.dec_corn3 = from bounding box
-    # l3_wcsinfo.dec_corn4 = from bounding box
-    # l3_wcsinfo.orientat = ???
-    # l3_wcsinfo.orientat_local = ???
+    l3_wcsinfo.pixel_shape = model.shape
+
+    pixel_center = [(v - 1) / 2.0 for v in model.shape[::-1]]
+    world_center = wcs(*pixel_center)
+    l3_wcsinfo.ra_center = world_center[0]
+    l3_wcsinfo.dec_center = world_center[1]
+    l3_wcsinfo.pixel_scale_local = compute_scale(wcs, world_center)
+    l3_wcsinfo.orientat_local = calc_pa(wcs, *world_center)
+
+    try:
+        footprint = utils.create_footprint(wcs, model.shape)
+    except Exception as excp:
+        log.warning("Could not determine footprint due to %s", excp)
+    else:
+        l3_wcsinfo.ra_corn1 = footprint[0][0]
+        l3_wcsinfo.ra_corn2 = footprint[1][0]
+        l3_wcsinfo.ra_corn3 = footprint[2][0]
+        l3_wcsinfo.ra_corn4 = footprint[3][0]
+        l3_wcsinfo.dec_corn1 = footprint[0][1]
+        l3_wcsinfo.dec_corn2 = footprint[1][1]
+        l3_wcsinfo.dec_corn3 = footprint[2][1]
+        l3_wcsinfo.dec_corn4 = footprint[3][1]
+        l3_wcsinfo.s_region = utils.create_s_region(footprint)
+
+    try:
+        l3_wcsinfo.x_ref = -transform["crpix1"].offset.value
+        l3_wcsinfo.y_ref = -transform["crpix2"].offset.value
+    except IndexError:
+        log.warning(
+            "WCS has no clear reference pixel defined by crpix1/crpix2. Assuming reference pixel is center."
+        )
+        l3_wcsinfo.x_ref = pixel_center[0]
+        l3_wcsinfo.y_ref = pixel_center[1]
+    world_ref = wcs(l3_wcsinfo.x_ref, l3_wcsinfo.y_ref)
+    l3_wcsinfo.ra_ref = world_ref[0]
+    l3_wcsinfo.dec_ref = world_ref[1]
+    l3_wcsinfo.pixel_scale = compute_scale(wcs, world_ref)
+    l3_wcsinfo.orientat = calc_pa(wcs, *world_ref)
+
+    try:
+        l3_wcsinfo.rotation_matrix = transform[
+            "pc_rotation_matrix"
+        ].matrix.value.tolist()
+    except Exception:
+        log.warning(
+            "WCS has no clear rotation matrix defined by pc_rotation_matrix. Calculating one."
+        )
+        l3_wcsinfo.rotation_matrix = utils.calc_rotation_matrix(l3_wcsinfo.orentat, 0.0)
```

Comment:
```suggestion
        l3_wcsinfo.rotation_matrix = utils.calc_rotation_matrix(l3_wcsinfo.orientat, 0.0)
```

Reply:
disturbing that got through all the checks. adding more checks...